s3fs/core.py

```diff
@@ -364,8 +364,11 @@ def _lsdir(self, path, refresh=False):
         if path not in self.dirs or refresh:
             try:
                 pag = self.s3.get_paginator('list_objects_v2')
+                config = {}
+                if max_items:
+                    config.update(MaxItems=max_items, PageSize=2 * max_items)
```

Comment:
If using max_items, the cache `self.dirs` should not be updated

Reply:
Fixed at https://github.com/dask/s3fs/pull/163/commits/61b73d35d580e3cd64beb72cece85508c35642f1